src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -48,9 +51,22 @@ internal class LaunchProfilesDebugLaunchProvider : DebugLaunchProviderBase, IDep
         /// <summary>
         /// Called by CPS to determine whether we can launch
         /// </summary>
-        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        public override async Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
         {
-            return TplExtensions.TrueTask;
+            try
+            {
+                // Launch providers to enforce requirements for debuggable projects
+                await QueryDebugTargetsInternalAsync(launchOptions, true);
+            }
+            catch(Exception e)
```

Comment:
>Exception [](start = 18, length = 9)

Just curious about this: should we throw a more specific exception type to avoid comparing exception messages? That way we don't return true if other unrelated exceptions are thrown (although that may be the desirable behavior in this case)

Reply:
While finding the fix I was checking all the exceptions. I agree. Changing this so I can get rid of the string comparer.

Reply:
The exception I am interested is a general exception with that specific message. Maybe I should create their own exception and catch that new exception.

Reply:
A new type of exception would be clearer. Comparing the exception message is not commonly done, though I can imagine cases where it's the only option. If you control the throw too, then a new exception would be good.

If we keep the string comparison, then if the comparison fails the exception should be rethrown (i.e. `else throw;`).